### PR TITLE
Add vm.GetResourceExtensions

### PR DESCRIPTION
**Pull request by paulmey**

### CODE REVIEW

File: management/virtualmachine/resourceextensions_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	"flag"
+	"testing"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/management/testutils"
+)
+
+func init() {
+	flag.BoolVar(&extraVerbose, "resourceextensions.verbose", false, "List all resource extensions")
```
Comment (ahmetalpbalkan) [inline management/virtualmachine/resourceextensions_test.go]: This is not good practice for test pkgs. test logs don't get printed unless -test.v exists. Make use of that if u can

File: management/virtualmachine/resourceextensions.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import (
+	"encoding/xml"
+)
+
+const (
+	azureResourceExtensionsURL = "services/resourceextensions"
+)
+
+func (self VirtualMachineClient) GetResourceExtensions() (extensions []ResourceExtension, err error) {
+	// from https://msdn.microsoft.com/en-us/library/azure/dn495441.aspx
```
Comment (ahmetalpbalkan) [inline management/virtualmachine/resourceextensions.go]: Move to godoc (see "storage")

File: management/virtualmachine/resourceextensions_test.go (hunk 1 above)
Comment (paulmey) [inline management/virtualmachine/resourceextensions_test.go]: I'd like to have an extra level of verbosity. Don't want to pollute the output with all resource extensions, but it is very handy to look things up. I can remove all of that code and people can put the `t.Logf("%+v", extension)` if they want it...

Comment (ahmetalpbalkan) [inline management/virtualmachine/resourceextensions_test.go]: @paulmey usually tests are ran without `-test.v`, and when it fails, all t.Log output is also printed (even without `-test.v`). That's why nobody probably needs an extra level of verbosity.

Comment (paulmey) [inline management/virtualmachine/resourceextensions_test.go]: No need for a lecture, I know all that. Likewise, if there was an extra level of verbosity, people would use it. I had fun figuring out how to add a flag, so I'm good. :-) Code is gone and this diff is outdated.

Comment (ahmetalpbalkan) [inline management/virtualmachine/resourceextensions_test.go]: It's just weird to have such undocumented params, especially for tests. That was mainly my comment.

Comment (paulmey) [inline management/virtualmachine/resourceextensions_test.go]: But it is very documented... that's exactly what the fourth parameter of [BoolVar](http://golang.org/pkg/flag/#BoolVar) does package does. Now merge it already if you're happy with 4440be0! :smile:

Comment (ahmetalpbalkan) [inline management/virtualmachine/resourceextensions_test.go]: @paulmey should I be running `go test . --help` to discover that flag? asking just out of curiosity.

Comment (paulmey) [inline management/virtualmachine/resourceextensions_test.go]: No, I haven't been able to get `go test` to list all flags for the package under test. You have to build the test binary with `go test -c .` and then execute the test binary with --help (`virtualmachine.test.exe --help`). I think this makes the ability for tests to declare extra flags very hard to discover and in most cases not the best practice (as you pointed out).

**Comment by ahmetalpbalkan** [comment]
LGTM :shipit:

**Pull request merged**